tests/testing_harness.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import print_function
+
+import filecmp
+import glob
+import hashlib
+from optparse import OptionParser
+import os
+from subprocess import Popen, STDOUT, PIPE, call
+import sys
+
+import numpy as np
+
+sys.path.insert(0, '../..')
+from openmc.statepoint import StatePoint
+from openmc.executor import Executor
+import openmc.particle_restart as pr
+
+
+class TestHarness(object):
+    """General class for running OpenMC regression tests."""
+    def __init__(self, statepoint_name, tallies_present=False):
+        self._sp_name = statepoint_name
+        self._tallies = tallies_present
+        self._opts = None
+        self._args = None
+
+    def execute_test(self):
+        """Run OpenMC with the appropriate arguments and check the outputs."""
+        self._parse_args()
+        try:
+            self._run_openmc()
+            self._test_output_created()
+            results = self._get_results()
+            self._write_results(results)
+            self._compare_results()
+        finally:
+            self._cleanup()
+
+    def _parse_args(self):
+        parser = OptionParser()
+        parser.add_option('--mpi_exec', dest='mpi_exec', action='store_true',
+                          default=False)
+        parser.add_option('--mpi_np', dest='mpi_np', default='3')
+        (self._opts, self._args) = parser.parse_args()
+
+    def _run_openmc(self):
+        if self._opts.mpi_exec:
+            mpi_procs = int(self._opts.mpi_np)
```

Comment:
We should avoid the cast to int by just reading in the option as an int in the first place, i.e., in _parse_args you'd have:
```python
parser.add_option('--mpi_np', dest='mpi_np', type=int, default=3)
```